src/main/kotlin/org/rust/ide/annotator/RustHighlightingAnnotator.kt

```diff
@@ -31,6 +35,15 @@ class RustHighlightingAnnotator : Annotator {
 
         override fun visitTypeParam(o: RustTypeParamElement) {
             holder.highlight(o, RustColor.TYPE_PARAMETER)
+            o.typeParamBounds?.let {
+                holder.highlight(it.colon, RustColor.KEYWORD) // feels off
```

Comment:
Year, as a general rule, it's better to avoid nested lambdas with implicit  parameters, or you may hit some [nasty bugs]. Also, using early returns helps to avoid deep nesting, so I probably would write something more like this:

```
val bounds = o.typeParamBouds ?: return
holder.highligh(bounds.colon, RustColor.KEYWORD)
```

Highlighting trait ref can be done in `visitTraitRef`.


[nasty bugs]: https://youtrack.jetbrains.com/issue/KT-11849